src/abstractdataframe/reshape.jl

```diff
@@ -399,3 +399,96 @@ function CategoricalArrays.CategoricalArray(v::RepeatedVector)
     res.refs = repeat(res.refs, inner = [v.inner], outer = [v.outer])
     res
 end
+
+
+Base.transpose(::AbstractDataFrame, args...; kwargs...) =
+    MethodError("`transpose` not defined for `AbstractDataFrame`s. Try `permutedims` instead")
+    # ↑ is 94 char, is that OK? (there are a couple other lines in this file that go over)
+
+"""
+    permutedims(df::AbstractDataFrame, src_namescol::ColumnIndex,
+                dest_namescol::Union{Symbol,Int};
+                makeunique::Bool=false)
+    permutedims(df::AbstractDataFrame, src_namescol::ColumnIndex; makeunique::Bool=false)
+    permutedims(df::AbstractDataFrame; makeunique::Bool=false)
+
+Turn a `DataFrame` on its side such that rows become columns
+and the column indexed by `src_namescol` becomes a header.
+In the resulting `DataFrame`,
+The header of `df` will become the first column
+with name specified by `dest_namescol`
+
+# Arguments
+- `df` : the `AbstractDataFrame`
+- `src_namescol` : the column that will become the new header.
```

Comment:
add information what are the requirements on values stored in this column (or that there are no requiremens)